Assignment6.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class Assignment6 {
+    /**
+     * Rearranges given to make it the same as desired.
+     *
+     * @param given   initial version of the array of numbers.
+     * @param desired final version of the array of numbers.
+     * @return list of moves made during arrangement.
+     */
+    public static List<List<Integer>> rearrange(int[] given, int[] desired) {
+        // Keys are values in given, values are their indexes in given.
+        Map<Integer, Integer> indexMap = new HashMap<>();
+        // Generated for testing purposes.
+        List<List<Integer>> listOfMoves = new LinkedList<>();
+
+        for (int i = 0; i < given.length; i++) {
+            indexMap.put(given[i], i);
+        }
+
+        for (int i = 0; i < desired.length; i++) {
+            // Already in its desired place.
+            if (indexMap.get(desired[i]) == i) {
+                continue;
+            }
+
```

Comment:
I guess we should test it if the current slot (i) should be the empty slot. In this case there should be no moves to be added. eg. {1, 2, 0, 3}->{0, 1, 3, 2} would generate something like car no. 0 moves which is not allowed